pkg/api/api.go

```diff
@@ -255,16 +256,17 @@ func (a *API) RegisterQuerier(queryable storage.Queryable, engine *promql.Engine
 	api := v1.NewAPI(
 		engine,
 		queryable,
-		querier.DummyTargetRetriever{},
-		querier.DummyAlertmanagerRetriever{},
+		func(context.Context) v1.TargetRetriever { return &querier.DummyTargetRetriever{} },
+		func(context.Context) v1.AlertmanagerRetriever { return &querier.DummyAlertmanagerRetriever{} },
 		func() config.Config { return config.Config{} },
 		map[string]string{}, // TODO: include configuration flags
 		v1.GlobalURLOptions{},
 		func(f http.HandlerFunc) http.HandlerFunc { return f },
-		func() v1.TSDBAdmin { return nil }, // Only needed for admin APIs.
-		false,                              // Disable admin APIs.
+		nil,   // Only needed for admin APIs.
+		"",    // TODO(codesome): what should this be?
```

Comment:
This is the directory where snapshots, created via the snapshot API, are stored. Snapshot API is disabled if admin API is disabled (we do disable it in the next line), so it's OK passing an empty string, but please add a comment to explain.